scripts/build_nim.sh

```diff
@@ -70,7 +70,7 @@ nim_needs_rebuilding() {
 	fi
 	# In case the local branch diverged and a fast-forward merge is not possible.
 	git fetch || true
-	git reset -q --hard origin/${NIM_COMMIT} || true
+	git reset -q --hard ${NIM_COMMIT} || true
```

Comment:
At a quick glance, it seems there are two remote definitions (origin and upstream). Wondering if removing the origin can break things in the future when NIM_COMMIT is back to a tag or branch? Like when having the same tag on both but with different reference commit, though unlikely.

```
origin	https://github.com/status-im/nimbus-build-system-desktop.git (fetch)
origin	https://github.com/status-im/nimbus-build-system-desktop.git (push)
upstream	https://github.com/status-im/nimbus-build-system.git (fetch)
upstream	https://github.com/status-im/nimbus-build-system.git (push)
```

Reply:
We probably want to specify the remote if we use a tag or branch. The rest of the script apart from this line requires only `NIM_COMMIT`, perhaps we could add an additional variable for specifying the remote that should only be assigned when using a branch name or a tag?

Reply:
Indeed the solution seems to be specifying a tag/branch. However, the current commit that we rely on (`121628357ec7fae91335bd392f03b0e6dd249319`) is not part of any branch or tag. The closest `version_1_2` has some differences that have to be tested first before jumping to it